lib/xgu/nv2a_regs.h

```diff
@@ -0,0 +1,1321 @@
+#       define NV097_SET_STENCIL_OP_V_INCR                        0x8507
+#       define NV097_SET_STENCIL_OP_V_DECR                        0x8508
+#   define NV097_SET_POLYGON_OFFSET_SCALE_FACTOR              0x00000384
+#   define NV097_SET_POLYGON_OFFSET_BIAS                      0x00000388
+#   define NV097_SET_FRONT_POLYGON_MODE                       0x0000038C
+#       define NV097_SET_FRONT_POLYGON_MODE_V_POINT               0x1B00
+#       define NV097_SET_FRONT_POLYGON_MODE_V_LINE                0x1B01
+#       define NV097_SET_FRONT_POLYGON_MODE_V_FILL                0x1B02
+#   define NV097_SET_BACK_POLYGON_MODE                        0x00000390
+#   define NV097_SET_CLIP_MIN                                 0x00000394
+#   define NV097_SET_CLIP_MAX                                 0x00000398
+#   define NV097_SET_CULL_FACE                                0x0000039C
+#       define NV097_SET_CULL_FACE_V_FRONT                         0x404
+#       define NV097_SET_CULL_FACE_V_BACK                          0x405
+#       define NV097_SET_CULL_FACE_V_FRONT_AND_BACK                0x408
+#   define NV097_SET_FRONT_FACE                               0x000003A0
+#       define NV097_SET_FRONT_FACE_V_CW                           0x900
+#       define NV097_SET_FRONT_FACE_V_CCW                          0x901
+#   define NV097_SET_NORMALIZATION_ENABLE                     0x000003A4
+#   define NV097_SET_LIGHT_ENABLE_MASK                        0x000003BC
+#           define NV097_SET_LIGHT_ENABLE_MASK_LIGHT0_OFF           0
+#           define NV097_SET_LIGHT_ENABLE_MASK_LIGHT0_INFINITE      1
+#           define NV097_SET_LIGHT_ENABLE_MASK_LIGHT0_LOCAL         2
+#           define NV097_SET_LIGHT_ENABLE_MASK_LIGHT0_SPOT          3
+#   define NV097_SET_TEXGEN_S                                 0x000003C0
+#       define NV097_SET_TEXGEN_S_DISABLE                         0x0000
+#       define NV097_SET_TEXGEN_S_EYE_LINEAR                      0x2400
+#       define NV097_SET_TEXGEN_S_OBJECT_LINEAR                   0x2401
+#       define NV097_SET_TEXGEN_S_SPHERE_MAP                      0x2402
+#       define NV097_SET_TEXGEN_S_REFLECTION_MAP                  0x8512
+#       define NV097_SET_TEXGEN_S_NORMAL_MAP                      0x8511
+#   define NV097_SET_TEXGEN_T                                 0x000003C4
+#   define NV097_SET_TEXGEN_R                                 0x000003C8
+#   define NV097_SET_TEXGEN_Q                                 0x000003CC
+#   define NV097_SET_TEXTURE_MATRIX_ENABLE                    0x00000420
+#   define NV097_SET_PROJECTION_MATRIX                        0x00000440
+#   define NV097_SET_MODEL_VIEW_MATRIX                        0x00000480
+#   define NV097_SET_INVERSE_MODEL_VIEW_MATRIX                0x00000580
+#   define NV097_SET_COMPOSITE_MATRIX                         0x00000680
+#   define NV097_SET_TEXTURE_MATRIX                 
```

Comment:
This `NV097_SET_LIGHT_*` stuff will be needed, if you are unsure, just have them accept `uint32_t value`; although some will be `vec3 = (float x, float y, float z)` or `vec4 = (float x, float y, float z, float w)`

Reply:
This comment had also moved. Comment indicates this was about `NV097_SET_LIGHT_`